relay-server/src/actors/processor.rs

```diff
@@ -59,13 +59,45 @@ use crate::envelope::{AttachmentType, ContentType, Envelope, Item, ItemType};
 use crate::extractors::RequestMeta;
 use crate::metrics_extraction::sessions::extract_session_metrics;
 use crate::metrics_extraction::transactions::extract_transaction_metrics;
-use crate::metrics_extraction::transactions::types::ExtractMetricsError;
+use crate::metrics_extraction::transactions::types::CommonTags;
+use crate::metrics_extraction::transactions::types::{
+    ExtractMetricsError, TransactionMeasurementTags, TransactionMetric,
+};
+use crate::metrics_extraction::IntoMetric;
 use crate::statsd::{RelayCounters, RelayHistograms, RelayTimers};
 use crate::utils::{
     self, get_sampling_key, log_transaction_name_metrics, ChunkedFormDataAggregator, FormDataIter,
     ItemAction, ManagedEnvelope, SamplingResult,
 };
 
+/// The length of a full measurement MRI, minus the name and the unit. This length is the same
+/// for every measurement-mri.
+///
+/// In 'fn process_measurements' we want to check if a measurement name is too long, to do that we
+/// check with a maximum length provided by the config, but that length represents the entire MRI name,
+/// and in process_measurements we only have access to the name part, and the unit part, so here
+/// we calculate the length of the remainder part. which isn't changing so we only need to do it once.
+static FIXED_MEASUREMENT_LEN: Lazy<isize> = Lazy::new(|| {
```

Comment:
Let's get rid of this. We can go through the path of making it configurable, or just hardcode it. This is harder to understand (let's optimize for ease of reading and understanding), and this logic may introduce bugs.

To configure it, we can do this dynamically from the server and get the information from the project config (like the `project_state.has_feature` examples below), or statically (the envelope processor has a property of relay config). Let's try to take the dynamic approach since it's easier to configure and update without new deployments.

Reply:
If we wanna hardcode it then we need to make sure the fixed part of measurement will not change in the future, since that would introduce bugs, about configuring it, we already configure how long an MRI but we still need to subtract the fixed part to calculate the name. 

we could do as jorris suggested, hardcore it in production code, but in tests ensure the FIXED_MEASUREMENT_LEN matches with the hardcoded value, I'm not sure if that would be simpler or not

Reply:
>1. If you hard-code this value, you need a unit test to detect a change in the naming schema.

Why? We could just hardcode a test that fails if the naming schema changes.

>2. If you make this value configurable, you allow the possibility of accepting measurements that do not fit into an MRI.

IIUC, that assumes the config value is incorrect.

>IMO this value is a derived value and should not be configurable.

I think I agree with that. I still don't understand the reason to not hardcode it though.

Reply:
> We could just hardcode a test that fails if the naming schema changes.

Yeah, that's what I mean. Something like `assert_eq!(FIXED_MEASUREMENT_LEN, len(mri) - len(name) - len(unit))`. This would also be fine with me. The one thing we should prevent is hardcoding this value without any checks if it corresponds to reality.

Reply:
@iker-barriocanal  the config value is the entire MRI, which is already configured, but we try to derive the name and unit value from the config value. You can't have both be configurable because one is a subset of the other

Reply:
> You can't have both be configurable because one is a subset of the other

Taking that as a requirement, I think you can make it configurable?

Reply:
you can make the max full mri length configurable, or the name of mri configurable, but not both. Currently full mri length is already configurable so it makes sense to just keep it like that currently. 

we cannot make both configurable because it's like an equation. 

mri_len == name_len + unit_len + fixed_len 

if we make both mri_len and name_len configurable then this equation cannot be guaranteed to be valid